src/core/featurelistmodel.cpp

```diff
@@ -101,6 +102,17 @@ void FeatureListModel::setCurrentLayer( QgsVectorLayer *currentLayer )
   emit currentLayerChanged();
 }
 
+QString FeatureListModel::currentLayerId() const
+{
+  return mCurrentLayer->id();
+}
+
+void FeatureListModel::setCurrentLayerId( const QString &layerId )
+{
+  QgsVectorLayer *layer = qobject_cast< QgsVectorLayer *>( QgsProject::instance()->mapLayer( layerId ) );
+  setCurrentLayer( layer );
+}
```

Comment:
QgsProject::mapLayer invokable has been merged.
This comment is just a reminder ;)